tutorials/nservicebus-101/lesson-4/tutorial.md

```diff
@@ -0,0 +1,212 @@
+
+This means that when the code for the credit card processing changes, we don't even need to touch (let alone test and redeploy) any of the code in the system except for that which is directly related to processing credit cards.
+
+
+## Defining events
+
+Creating an event message is just similar to creating an event. We just create a class and mark it with the `IEvent` (rather than `ICommand`) marker interface.
+
+snippet:Event
+
+All the other considerations for command messages apply to events as well. Properties can be simple types, complex types, or collections - whatever the message serializer supports.
+
+With events, you should be even more careful to refrain from getting carried away and putting too much information in an event message. Sometimes this complexity can't be avoided for commands, as the command receiver needs the information to do its job. That's manageable for commands because the sender and receiver are highly coupled anyway. For events, it's a different story. Since a publisher of an event does not know (or care) how many subscribers it has, it may not be possible to modify all of them if a change is required to the event.
+
+
+## Handling events
+
+Create a handler class by implementing `IHandleMessages<T>` where `T` is the type of the event message.
+
+snippet:EventHandler
+
+
+## Subscribing to events
+
+For the MSMQ transport, NServiceBus needs to know which endpoint is responsible for publishing an event, so that it can send the publishing endpoint a subscription request message.
+
+You can configure the publisher endpoint via the Routing API like this:
+
+snippet:RegisterPublisher
+
+NOTE: Some other transports have built-in publish/subscribe capabilities, so all that is required to subscribe to an event is to create a message handler for the event. A subscription request message is not required. Because the routing configuration is scoped to the transport type, transports that don't require publisher configuration simply won't contain this API.
+
+
+## Exercise
+
+Now that we've learned about events and the Publish/Subscribe pattern, let's make use of it in our ordering system. When a user places an order, we're going to publish an OrderPlaced event, and then subscribe to it from two brand new endpoints: Billing and Shipping.
+
+We'll also create a new OrderBilled event that will be published by the Billing endpoint once the credit card transaction is complete.
+
+![Lesson 4 Diagram](diagram.png)
+
+When the Shipping endpoint receives both the OrderPlaced and OrderBilled, it will know that it is time to ship the product to the customer. (Because this requires stored state, we can't accomplish that with message handlers alone. To implement that functionality, we would need a [Saga](/nservicebus/sagas/), but that will not be covered in this lesson.)
```

Comment:
> (Because this requires stored state, we can't accomplish that with message handlers alone. To implement that functionality, we would need a [Saga](/nservicebus/sagas/), but that will not be covered in this lesson.)

the brackets are redundant